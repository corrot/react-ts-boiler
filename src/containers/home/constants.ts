export const INCREMENT: string = 'INCREMENT';
export const DECREMENT: string = 'DECREMENT';
export const INCREMENT_ASYNC: string = 'INCREMENT_ASYNC';
export const DECREMENT_ASYNC: string = 'DECREMENT_ASYNC';
