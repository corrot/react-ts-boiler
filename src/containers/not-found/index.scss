.c-not-found {
	font-family: monospace;
	font-size: 3rem;
	line-height: 1.5;
	color: red;
	text-align: center;
}
