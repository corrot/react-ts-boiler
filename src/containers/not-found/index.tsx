import * as React from 'react';

import './index.scss';

export const NotFound = (): React.ReactElement<any> => (
	<div className="c-not-found">
		404 <br />
		Page not found
	</div>
);

export default NotFound;
