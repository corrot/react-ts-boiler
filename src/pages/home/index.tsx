import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';

import Api from '@/services/api';
import { Flex } from '@/styled/flex';
import { IUser } from '@/types/models/user';
import { Button } from '@/components/library/button';

export const Home = () => {
	const { t } = useTranslation();
	const [users, setUsers] = useState<IUser[]>([]);
	const fetchUsers = async () => {
		const { data } = await Api.user.getUsers();
		setUsers(data);
	};

	return (
		<Flex center>
			<Flex center sm={6} xs={8} padding="1rem">
				<Button onClick={fetchUsers}>{t('Main page')}</Button>
			</Flex>
			<Flex direction="column" center sm={6} xs={8} padding="1rem">
				{users.map(u => (
					<div key={u.id}>{u.username}</div>
				))}
			</Flex>
		</Flex>
	);
};

export default Home;
