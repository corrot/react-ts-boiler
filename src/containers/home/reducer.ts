import { INCREMENT, DECREMENT } from './constants';

export interface CounterState {
	count: number;
}

export interface CounterAction {
	type: string;
	payload: number;
}

export const homeInitialState: CounterState = {
	count: 0
};

export default (state = homeInitialState, { type, payload }: CounterAction) => {
	switch (type) {
		case INCREMENT:
		case DECREMENT:
			return {
				...state,
				count: state.count + payload
			};

		default:
			return state;
	}
};
