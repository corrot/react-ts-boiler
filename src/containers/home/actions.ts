import { CounterAction } from './reducer';
import { INCREMENT, DECREMENT } from './constants';

export const increment = (): CounterAction => ({
	type: INCREMENT,
	payload: 1
});

export const decrement = (): CounterAction => ({
	type: DECREMENT,
	payload: -1
});
