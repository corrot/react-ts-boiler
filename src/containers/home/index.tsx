import * as React from 'react';
import SVG from 'react-inlinesvg';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import Button from '@components/button';
import { RootStore } from '@src/store';
import { CounterState } from './reducer';
import { increment, decrement } from './actions';
import { INCREMENT_ASYNC, DECREMENT_ASYNC } from './constants';

import './index.scss';

import * as ReduxSagaLogo from '@assets/redux-saga.png';

interface Counter {
	count: number;
}

interface Props {
	readonly counter: Counter;
	readonly dispatch: Dispatch;
}

export const Home = ({ counter, dispatch }: Props): React.ReactElement<any> => (
	<React.Fragment>
		<SVG src="assets/react.svg" className="c-svg-icon" />

		<div className="o-wrapper">
			<Button onClick={() => dispatch(decrement())}>-</Button>

			<Button onClick={() => dispatch({ type: DECREMENT_ASYNC })}>Async -</Button>

			<small>{counter.count}</small>

			<Button onClick={() => dispatch({ type: INCREMENT_ASYNC })}>Async +</Button>

			<Button onClick={() => dispatch(increment())}>+</Button>
		</div>

		<img src={ReduxSagaLogo as any} alt="Redux Saga Logo" style={{ display: 'block', margin: 'auto' }} />
	</React.Fragment>
);

export default connect(
	(
		store: RootStore
	): {
		readonly counter: CounterState;
	} => ({
		counter: store.counter
	})
)(Home);
