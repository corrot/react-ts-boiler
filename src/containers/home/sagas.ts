import {
	put,
	delay,
	takeEvery,
	SimpleEffect,
	PutEffectDescriptor,
	CallEffectDescriptor,
	ForkEffectDescriptor
} from 'redux-saga/effects';

import { CounterAction } from './reducer';
import { increment, decrement } from './actions';
import { INCREMENT_ASYNC, DECREMENT_ASYNC } from './constants';

export function* incrementAsync(): IterableIterator<
	SimpleEffect<'PUT', PutEffectDescriptor<CounterAction>> | SimpleEffect<'CALL', CallEffectDescriptor>
> {
	yield delay(1000);
	yield put(increment());
}

export function* decrementAsync(): IterableIterator<
	SimpleEffect<'PUT', PutEffectDescriptor<CounterAction>> | SimpleEffect<'CALL', CallEffectDescriptor>
> {
	yield delay(1000);
	yield put(decrement());
}

export function* incrementSaga(): IterableIterator<SimpleEffect<'FORK', ForkEffectDescriptor>> {
	yield takeEvery(INCREMENT_ASYNC, incrementAsync);
}

export function* decrementSaga(): IterableIterator<SimpleEffect<'FORK', ForkEffectDescriptor>> {
	yield takeEvery(DECREMENT_ASYNC, decrementAsync);
}
