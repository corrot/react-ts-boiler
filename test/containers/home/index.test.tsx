import React from 'react';
import renderer from 'react-test-renderer';

import { Home } from '@/containers/home';

describe('Home component', () => {
	it('should render successfully', () => {
		const tree = renderer.create(<Home />).toJSON();

		expect(tree).toMatchSnapshot();
	});
});
