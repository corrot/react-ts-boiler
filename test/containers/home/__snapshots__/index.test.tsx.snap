// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render successfully 1`] = `
<main>
  <button
    className="sc-bdVaJa ldXAlL"
    onClick={[Function]}
  >
    -
  </button>
  <small>
    0
  </small>
  <button
    className="sc-bdVaJa ldXAlL"
    onClick={[Function]}
  >
    +
  </button>
</main>
`;
